### PR TITLE
rewritten API in Promise, brought flexible error handling, created new reuasble cardComponent in hackathon cards, hackathonCreate date Validation

### PR DESCRIPTION
**Pull request by rajatsandeepsen**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] HotFix (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

**Comment by rajatsandeepsen** [comment]
link for testing [mulearn.vercel.app](https://mulearn.vercel.app/)

File: src/App.tsx
Comment (Jenin82) [inline src/App.tsx:1]: don't change the path in app.tsx

**Review by Jenin82 — CHANGES_REQUESTED** [review]

File: src/App.tsx
Comment (rajatsandeepsen) [reply src/App.tsx:1]: Aswin paranju 😥

**Review by Jenin82 — APPROVED** [review]
ok to merge

**Review by AswinAsok — APPROVED** [review]

**Pull request merged**